lib/features/poscan_putobject/presentation/widgets/index_input_field.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:threedpass/core/widgets/input/textformfield/textformfield.dart';
+
+class IndexInputField extends StatefulWidget {
+  const IndexInputField({
+    required this.controller,
+    final Key? key,
+  }) : super(key: key);
+
+  final TextEditingController controller;
+
+  @override
+  State<StatefulWidget> createState() => _IndexInputFieldState();
+}
+
+class _IndexInputFieldState extends State<IndexInputField> {
+  static final numericInputFormatter = FilteringTextInputFormatter.allow(RegExp(r'[0-9]')); // Only allow decimal digits
+
+  void onClearPressed(final BuildContext context) {
+    widget.controller.clear();
+  }
+
+  @override
+  Widget build(final BuildContext context) {
+    return D3pTextFormField(
+      labelText: 'Id'.tr(),
+      hintText: 'Id'.tr(),
+      keyboardType: const TextInputType.numberWithOptions(decimal: true),
+      suffixButton: Icons.clear,
+      onSuffixButtonPressed: () => onClearPressed(context),
```

Comment:
I think suffix buttons are redudant in this dialog. Could you remove them from every input?